packages/minimongo/minimongo.js

```diff
@@ -544,6 +544,16 @@ LocalCollection.prototype.insert = function (doc, callback) {
   var self = this;
   doc = EJSON.clone(doc);
 
+  // Make sure field names do not contain dots to line up with Mongo's
+  // field name restrictions:
+  // https://docs.mongodb.com/manual/reference/limits/#Restrictions-on-Field-Names
+  if (doc) {
+    const invalidFieldMatches = JSON.stringify(doc).match(/"([^"]*\.[^"]*)":/);
```

Comment:
I believe this RegEx might be susceptible to match inadvertently on the `value` side.  

Consider the JSON:

```json
{"value":"silly quoted JSON in here \"︵ (°.°)/ ︵\": inadvertently matches!  eeps"}
```

...which I believe would throw the error incorrectly.

As an alternative, thoughts on using the [`replacer` function](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) of `JSON.stringify`?

```js
JSON.stringify(doc, (key) => {
  if (key.indexOf(".") > -1) {
    throw MinimongoError(`Field ${key} must not contain '.'`);
  }
  return value;
});
```

We could fix the regex, but I'm just thinking that the built-in function might be faster anyway and more able to handle whatever is thrown at it.